### DISCUSSION TITLE
Update many with filter on related entities

### DISCUSSION THREAD

**Discussion by TheCataliasTNT2k** [SeaQL/sea-orm#1406] [Q&A] [answered]
Hey,
there is an [example](https://www.sea-ql.org/SeaORM/docs/0.4.x/basic-crud/update/#update-many) how to update many entities at once and add a filter to that update. Is there a way to filter by an attribute, which is only existing in a related table?
Using your [cake example](https://www.sea-ql.org/SeaORM/docs/relation/bakery-schema/), let's say I want to

`update` the `price` for all `cakes`
which are `related` to a `bakery` with a specific `name`

Say we want to set the price to zero, because the bakery has anniversary or something.

This should be done completely in SQL, I do not want to fetch the id of the bakery before or something like that.
My real case is a little bit more complex, so have a look at the following example:

`update` the `total` for all `lineitem`s
which are related to an `order` with a specific `customer_id`

Say we want to set the price to 0, because we like the customer or something.

As you can see, there can be multiple orders involved, in contrast to the bakery example above.
What would this look like in code?

**Comment by TheCataliasTNT2k** [comment, marked as answer]
I found a solution if you only need the relations for filtering and not for setting the new value:
```rust
// Entity is lineitem's entity
Entity::update_many()
.filter(
    Column::OrderId.in_subquery(
        sea_query::Query::select()
            .from(order::Entity)
            .column(order::Column::Id)
            .and_where(order::Column::CustomerId.eq(customer_id))
            .to_owned(),
    )
)
.col_expr(Column::Total, Expr::value(0))
.exec(db)
.await
```
This is not testes though...

**Reply by billy1624** [reply]
This is correct!

**Reply by TheCataliasTNT2k** [reply]
Yes, this works for my simple example above, but not for more complicated things, as I already mentioned.
